test/question.spec.js

```diff
@@ -0,0 +1,196 @@
+  describe('Positive Tests', () => {
+    let token;
+    let question_id;
+    const user = {
+      username: 'kufre',
+      firstname: 'Kufre',
+      lastname: 'Okon',
+      email: `${Date.now()}_example@example.com`,
+      password: 'test123',
+    };
+    before(async () => {
+      const password = await hashPassword('test123');
+      await createUser({ ...user, password });
+      const response = await request(app)
+        .post('/auth/login')
+        .send(user)
+        .set('Accept', 'application/json');
+      token = response.body.accessToken;
+    });
+    it('Should create post a new question', async () => {
+      const response = await postRequest('/questions', token)
+        .send({
+          question_text: 'question',
+        })
+        .expect(200);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('message');
+      expect(resp_data).to.have.property('question');
+      expect(resp_data.message).to.be.an('string');
+      expect(resp_data.message).to.equal('Successfully posted question');
+      expect(resp_data.question).to.be.an('object');
+      question_id = resp_data.question.id;
+    });
+    it('Should edit question', async () => {
+      const response = await putRequest(`/questions/${question_id}`, token)
+        .send({
+          question_text: 'question_update',
+        })
+        .expect(200);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('message');
+      expect(resp_data).to.have.property('question');
+      expect(resp_data.message).to.be.an('string');
+      expect(resp_data.message).to.equal('Successfully edited question');
+      expect(resp_data.question).to.be.an('object');
+    });
+    it('Should search question', async () => {
+      const search_query = 'question';
+      const response = await getRequest(`/questions/search/${search_query}?page=0&limit=10`, token).expect(200);
+
+      const resp_data = response.body;
+      expect(resp_data).to.be.an('object');
+      expect(resp_data).to.have.property('message');
+      expect(resp_data).to.have.property('questions');
+      expect(resp_data.message).to.be.an('string');
+      expect(resp_data.message).to.equal('Successfully got questions.');
+      expect(resp_data.questions).to.be.an('array');
+    });
+    it('Should get question', async () => {
+      const question = await fetchQuestion({ id: question_id }, true);
+
+      expect(question).to.be.an('object');
+      expect(question).to.have.property('question_text');
+    });
+    it('Should get questions', async () => {
+      const questions = await fetchQuestions(true);
+
+      expect(questions).to.be.an('array');
+    });
+    it('Should get questions with most answers', async () => {
+      const questions = await fetchQuestionsWithMostAnswers(question_id);
+      expect(questions).to.be.an('array');
+    });
```

Comment:
Also check that the array is not empty

Reply:
It might be empty though because it's possible there hasn't yet been any answers yet